modules/sdk-coin-sol/src/lib/transaction.ts

```diff
@@ -461,4 +477,47 @@ export class Transaction extends BaseTransaction {
       durableNonce: durableNonce,
     };
   }
+
+  private explainRawMsgAuthorizeTransaction(): TransactionExplanation {
```

Comment:
if i added an extra ix, this fn would not display it. I understand that your other methods would hopefully throw an error upstream if that where the case, but in isolation, it wouldnt catch this issue. Just something to be mindful of. 

Ideally decode/explain functions take into account the entire tx without assuming context of upstream validation.

Reply:
Also seems you might be missing tests for `explainTransaction` in this scenario? Maybe i missed them.